reascripts/common/libs/PathUtil.lua

```diff
@@ -6,6 +6,11 @@
 
 PathUtil = {}
 
+-- Returns true if the given file path has an extension, false otherwise.
+PathUtil.has_extension = function(filepath)
+  return filepath:find("%.[^%.\\/%s]*$") and true or false
```

Comment:
Do we consider whitespace at the end of a filename to be... no longer an extension? I wonder if we need to strip whitespace from filenames in the UI handling, since we allow arbitrary user input.

Reply:
a space is technically a valid filename character, right? asking this out loud even though i know that the answer is technically true. mildly annoyed atm at the notion of a file extension generally since in most contexts it's not much more than a potentially unreliable hint lol